includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -95,11 +95,74 @@ public function getData() {
 		return $data;
 	}
 
-	private function getHref( $hrefKey, $protocolRelative = false ) {
+	protected function getDomainForCityId( $cityId ) {
+		global $wgServer;
+		if ( $cityId === null ) {
+			return $wgServer;
+		} else {
+			return  WikiFactory::cityIDtoUrl( $cityId );
+		}
+	}
+
+	protected function getDevDomain() {
+		global $wgDevDomain, $wgDevelEnvironment;
+		if ( !$wgDevelEnvironment || empty( $wgDevDomain ) ) {
+			return false;
+		}
+
+		return $wgDevDomain;
+	}
+
+	/**
+	 * Function will substitute primary domain in the given url to the domain given (or current)
+	 * wiki is setup for (i.e. wikia.com or fandom.com). This will allow to have proper domains in the urls when
+	 * link is stored as .wiki.com but the wiki is running on other domain.
+	 *
+	 * @example https://www.wikia.com/signin -> https://www.fandom.com/signin
+	 *
+	 * @param $url string source url to be updated
+	 * @param null $cityId int id of the wiki for the base domain used in the substitution (null will use current wiki)
+	 * @return string final url with base domain substituted (or original url in case of some errors)
+	 */
+	public function useWikiBaseDomainInUrl( $url, $cityId = null ) {
```

Comment:
Is `$cityId` needed? I don't see anything using it, so it seems like it might be better to remove it and add it back if there's ever a need for it.

Reply:
OK.  I'll remove it